### PR TITLE
Fix bug in getting highest confirmed nonce

### PR DESCRIPTION
**Pull request by bolekk**
Fixing a SQL bug that searches for a global max nonce instead of a key-scoped one. Because of that, some transactions can be incorrectly put in confirmed_missing_receipt state.

See story and attached gdoc for detailed incident analysis.

### CODE REVIEW

File: core/chains/evm/txmgr/eth_confirmer_test.go [hunk 1]
```diff
@@ -680,6 +695,68 @@ func TestEthConfirmer_CheckForReceipts_should_not_check_for_likely_unconfirmed(t
...
+
+	state1, fromAddress1 := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
+	state2, fromAddress2 := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
+
+	ethClient := cltest.NewEthClientMockWithDefaultChain(t)
+	ethClient.On("NonceAt", mock.Anything, mock.Anything, mock.Anything).Return(uint64(20), nil)
+	evmcfg := evmtest.NewChainScopedConfig(t, cfg)
+
+	ec := cltest.NewEthConfirmer(t, db, ethClient, evmcfg, ethKeyStore, []ethkey.State{state1, state2}, nil)
+	ctx := context.Background()
```
Comment (pinebit) [inline core/chains/evm/txmgr/eth_confirmer_test.go:714]: For context in tests, please use `testutils.Context(t)`

File: core/chains/evm/txmgr/eth_confirmer_test.go [hunk 2]
```diff
@@ -680,6 +695,68 @@ func TestEthConfirmer_CheckForReceipts_should_not_check_for_likely_unconfirmed(t
...
+
+	db := pgtest.NewSqlxDB(t)
+	cfg := configtest.NewTestGeneralConfig(t)
+	borm := cltest.NewTxmORM(t, db, cfg)
+	ethKeyStore := cltest.NewKeyStore(t, db, cfg).Eth()
+
+	state1, fromAddress1 := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
+	state2, fromAddress2 := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
+
+	ethClient := cltest.NewEthClientMockWithDefaultChain(t)
```
Comment (pinebit) [inline core/chains/evm/txmgr/eth_confirmer_test.go:709]: `// Deprecated: use evmtest.NewEthClientMockWithDefaultChain`

**Review by pinebit — CHANGES_REQUESTED** [review]

**Comment by bolekk** [comment]
Extended the test case to also cover multiple chains (apart from multiple keys on the same chain).

**Review by pinebit — APPROVED** [review]
LGTM

**Review by reductionista — APPROVED** [review]
LGTM, assuming the smoke test failures are just flakes

**Comment by bolekk** [comment]
Fix bug in getting highest confirmed nonce

**Pull request merged**